pkg/git/Util.go

```diff
@@ -81,8 +71,57 @@ func GetUserNamePassword(gitProvider *sql.GitProvider) (userName, password strin
 	case sql.AUTH_MODE_ANONYMOUS:
 		return "", "", nil
 	case sql.AUTH_MODE_SSH:
-		return "", "", fmt.Errorf("ssh not supported")
+		return "", "", nil
 	default:
 		return "", "", fmt.Errorf("unsupported %s", gitProvider.AuthMode)
 	}
 }
+
+func GetOrCreateSshPrivateKeyOnDisk(gitProviderId int, sshPrivateKeyContent string) (privateKeyPath string, err error) {
```

Comment:
we should update the data, so that update in  ssh key gets reflected

Reply:
When we update gitProvider - then sshKey gets updated on the disk. 
This particular flow just get the key from disk if exist, if not exists then it creates.